src/native/corehost/fxr/command_line.cpp

```diff
@@ -164,6 +164,17 @@ namespace
                 trace::verbose(_X("Application '%s' is not a managed executable."), app_candidate.c_str());
                 if (!exec_mode)
                 {
+                    // Check if this is a non-managed file with directory separator that exists
+                    // This should show a specific error instead of routing to CLI
+                    bool has_dir_separator = app_candidate.find(DIR_SEPARATOR) != pal::string_t::npos;
+                    if (has_dir_separator)
+                    {
+                        if (pal::file_exists(app_candidate))
+                        {
+                            trace::error(_X("The application '%s' is not a managed .dll."), app_candidate.c_str());
+                            return StatusCode::InvalidArgFailure;
```

Comment:
We were actually relying on the host to forward these to the CLI, we handle `dotnet ./script.cs` to run the `./script.cs` file-based app.

Reply:
Definitely agree here - folks are using this with unix shebangs and that's been a big driver for the file-based app feature set.